connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -335,43 +340,51 @@ object KustoWriter {
-      case StringType => GetStringFromUTF8(row, fieldIndexInRow, nested)
+      case StringType => GetStringFromUTF8(row.getUTF8String(fieldIndexInRow), nested)
       case BooleanType => row.getBoolean(fieldIndexInRow).toString
-      case _: StructType => convertStructToCsv(row.getStruct(fieldIndexInRow, dataType.asInstanceOf[StructType].length), dataType.asInstanceOf[StructType], dateFormat)
-      case _: ArrayType => convertArrayToCsv(row.getArray(fieldIndexInRow), dataType, dateFormat)
+      case strType: StructType => convertStructToCsv(row.getStruct(fieldIndexInRow, dataType.asInstanceOf[StructType].length), strType, dateFormat)
+      case arrType: ArrayType => convertArrayToCsv(row.getArray(fieldIndexInRow), arrType.elementType, dateFormat)
+      case mapType: MapType => convertMapToCsv(row.getMap(fieldIndexInRow), mapType, dateFormat)
       case _ => row.get(fieldIndexInRow, dataType).toString
     }
   }
 
   private def convertStructToCsv(row: InternalRow, schema: StructType, dateFormat: FastDateFormat): String = {
     val fields = schema.fields
-    val result:scala.collection.mutable.Map[String,String] = scala.collection.mutable.Map()
+    val result: scala.collection.mutable.Map[String, String] = scala.collection.mutable.Map()
 
     for (x <- fields.indices) {
-      result(fields(x).name) = getField(row, x, fields(x).dataType, dateFormat, nested=true)
+      result(fields(x).name) = getField(row, x, fields(x).dataType, dateFormat, nested = true)
     }
-
-    "{" + result.map{case (k, v) => "\"" +  k + "\"" + ":" + v }.mkString(",") + "}"
+    "{" + result.map { case (k, v) => "\"" + k + "\"" + ":" + v }.mkString(",") + "}"
   }
 
   private def convertArrayToCsv(ar: ArrayData, fieldsType: DataType, dateFormat: FastDateFormat): String = {
     if (ar == null) null else {
       if (ar.numElements() == 0) "[]" else {
-        val result:Array[String] = new Array(ar.numElements())
-        val fieldType = fieldsType.asInstanceOf[ArrayType].elementType
-        for (x <- 0 until ar.numElements()) {
-          result(x) = getField(ar, x, fieldType, dateFormat, nested=true)
-        }
 
-        "[" + result.mkString(",") + "]"
+        "[" + convertArrayToCsvImpl(ar, fieldsType, dateFormat) + "]"
       }
     }
   }
+  private def convertArrayToCsvImpl(ar: ArrayData, fieldsType: DataType, dateFormat: FastDateFormat, nested: Boolean = true): String = {
+    val result: Array[String] = new Array(ar.numElements())
+    for (x <- 0 until ar.numElements()) {
+      result(x) = getField(ar, x, fieldsType, dateFormat, nested)
+    }
+    result.mkString(",")
+  }
+
+
+  private def convertMapToCsv(map: MapData, fieldsType: MapType, dateFormat: FastDateFormat): String = {
+    val a = convertArrayToCsvImpl(map.keyArray(), fieldsType.keyType, dateFormat, nested = false)
+    val b = convertArrayToCsvImpl(map.valueArray(), fieldsType.valueType, dateFormat)
+    "{" +"\"" + a + "\"" + ":"+ b + "}"
```

Comment:
This logic is not correct.
Let's say we have map.keyArray() = ["a","b"] and map.valueArray() = [1,2]
So we get:
a = "a","b" 
b = 1,2
and the result of this method is:
{""a","b" ":1,2}

What we want is:
{"a":1,"b":2}